content/chainguard/chainguard-images/using-the-image-diff-api.md

```diff
@@ -0,0 +1,220 @@
+2. The next tool is `crane`, which you will use to interact with the Chainguard Registry. Follow the [crane installation](https://github.com/google/go-containerregistry/blob/main/cmd/crane/README.md#installation) guide that matches your operating system.
+3. You will also need `curl` or a programming language of your choice to make authenticated API calls. There is an [example using Go](#making-a-request-programmatically-to-the-diff-api-using-go) at the end of this guide.
+4. Finally, the `jq` is useful to parse and filter the resulting JSON output from the API on the command line.
+
+With these tools in place, you can begin gathering the information that you need to authenticate to and query the Diff API.
+
+## Authenticating and Getting a Token
+
+To get started, ensure you are authenticated to the Registry using the `chainctl auth login` command. Then locate your authentication token and create a shell variable for it.
+
+If you are using Linux, run the following:
+
+```sh
+TOKEN=$(cat ~/.cache/chainguard/https\:--console-api.enforce.dev/oidc-token)
+```
+
+If you are using macOS, run the following:
+
+```sh
+TOKEN=$(cat ~/Library/Caches/chainguard/https:--console-api.enforce.dev/oidc-token)
+```
+
+Note: if at any point in this tutorial you receive an error like the following, your token has expired. Login and set your `$TOKEN` shell variable again:
+
+```
+{"code":16, "message":"failed to verify token", "details":[]}
+```
+
+For the purposes of this guide, we will use the `public-images` Chainguard IAM group. Set an environment variable corresponding to the group's [UIDP](/chainguard/administration/cloudevents/events-reference/#uidp-identifiers), which will be used later on:
+
+```sh
+GROUP_UIDP=$(chainctl iam group describe public-images -ojson |jq -r '.id')
+```
+
+If you are using a private registry, be sure to substitute in your IAM group name in place of `public-images` to use your specific private group UIDP identifier in your subsequent commands.
+
+## Getting Image Digests to Compare
+
+Now you will need to find image digests to compare. You can compare any two arbitrary digests using the API. In this example we will use the public `cgr.dev/chainguard/nginx` repository and find digests for the `latest` and `latest-dev` tags. You can also compare image digests for the same tag by combining data from the [Tag History API](/chainguard/chainguard-images/using-the-tag-history-api/).
+
+The first digest in this example will be referred to as the `FROM_DIGEST`, and the second comparison image as `TO_DIGEST`.
+
+Begin by creating a shell variable to hold the repository URL:
+
+```sh
+REPO_URL="cgr.dev/chainguard/nginx"
+```
+
+Note: if you are using a private registry, replace `cgr.dev/chainguard/nginx` with your relevant private Chainguard Registry and repository name as required.
```

Comment:
```suggestion
If you are using a private registry, be sure to replace `cgr.dev/chainguard/nginx` with your relevant private Chainguard Registry and repository name as required.
```

I think it looks nicer if we only use "Note:" in places where we do a proper note block (with a `>` preceding it)